### PR TITLE
Access token grant handler

### PR DESCRIPTION
**Pull request by janakamarasena**
Added token validation classes to the same bundle

### CODE REVIEW

**Review by ayyoob — APPROVED** [review]

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/OAuthValidationResponse.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    public boolean isValid() {
+        return isValid;
+    }
+
+    public void setIsValid(boolean isValid) {
+        this.isValid = isValid;
+    }
+}
```
Comment (Kanapriya) [inline components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/OAuthValidationResponse.java:59]: Add new line

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorDataHolder.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            throw new IllegalStateException("OAuth2TokenValidation service is not initialized properly");
+        }
+        return oAuth2TokenValidationService;
+    }
+
+    public void setOAuth2TokenValidationService(
+            OAuth2TokenValidationService oAuth2TokenValidationService) {
+        this.oAuth2TokenValidationService = oAuth2TokenValidationService;
+    }
+}
```
Comment (Kanapriya) [inline components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorDataHolder.java:49]: Add new line

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorServiceComponent.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     * @param tokenValidationService An instance of OAuth2TokenValidationService
+     */
+    @SuppressWarnings("unused")
+    protected void unsetOAuth2ValidationService(OAuth2TokenValidationService tokenValidationService) {
+        if (log.isDebugEnabled()) {
+            log.debug("Unsetting OAuth2TokenValidationService Service");
+        }
+        OAuthAuthenticatorDataHolder.getInstance().setOAuth2TokenValidationService(null);
+    }
+}
```
Comment (Kanapriya) [inline components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorServiceComponent.java:76]: Add new line

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/OAuthValidationResponse.java (hunk 1 above)
Comment (janakamarasena) [reply components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/OAuthValidationResponse.java:59]: Fixed with https://github.com/wso2/carbon-device-mgt/pull/1278

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorDataHolder.java (hunk 2 above)
Comment (janakamarasena) [reply components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorDataHolder.java:49]: Fixed with https://github.com/wso2/carbon-device-mgt/pull/1278

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorServiceComponent.java (hunk 3 above)
Comment (janakamarasena) [reply components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/internal/OAuthAuthenticatorServiceComponent.java:76]: Fixed with https://github.com/wso2/carbon-device-mgt/pull/1278

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    private static final String BEARER_TOKEN_TYPE = "bearer";
+
+    /**
+     * This method gets a string accessToken and validates it and generate the OAuth2ClientApplicationDTO
+     * containing the validity and user details if valid.
+     *
+     * @param token which need to be validated.
+     * @return OAuthValidationResponse with the validated results.
+     */
+    public OAuthValidationResponse validateToken(String token) throws RemoteException{
```
Comment (omindu) [inline components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java:40]: Missing whitespace after `RemoteException`

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        OAuth2TokenValidationRequestDTO validationRequest = new OAuth2TokenValidationRequestDTO();
+        OAuth2TokenValidationRequestDTO.OAuth2AccessToken accessToken =
+                validationRequest.new OAuth2AccessToken();
+        accessToken.setTokenType(BEARER_TOKEN_TYPE);
+        accessToken.setIdentifier(token);
+        validationRequest.setAccessToken(accessToken);
+        OAuth2TokenValidationResponseDTO tokenValidationResponse = OAuthAuthenticatorDataHolder.getInstance().
+                getOAuth2TokenValidationService().findOAuthConsumerIfTokenIsValid(validationRequest).getAccessTokenValidationResponse();
```
Comment (omindu) [inline components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java:48]: Check the line length

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java (hunk 4 above)
Comment (janakamarasena) [reply components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java:40]: Fixed with https://github.com/wso2/carbon-device-mgt/pull/1278

File: components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java (hunk 5 above)
Comment (janakamarasena) [reply components/identity-extensions/org.wso2.carbon.device.mgt.oauth.extensions/src/main/java/org/wso2/carbon/device/mgt/oauth/extensions/handlers/grant/oauth/validator/LocalOAuthValidator.java:48]: Fixed with https://github.com/wso2/carbon-device-mgt/pull/1278

**Pull request merged**